assets/src/domain/eventEditor/ui/datetimes/hooks/useDatesActionMenuHandler.tsx

```diff
@@ -5,32 +5,28 @@ import EditDateButton from '../datesList/actionsMenu/EditDateButton';
 import DeleteDateButton from '../datesList/actionsMenu/DeleteDateButton';
 import AssignTicketsButton from '../datesList/actionsMenu/AssignTicketsButton';
 import { Datetime } from '@edtrServices/apollo/types';
-import { EntitySubscriptionCallback } from '@appLayout/entityActionsMenu';
-import { AdditionalDateMenuOptions } from '../types';
+import { EntityActionsSubscriptionCb } from '@appLayout/entityActionsMenu';
 import { useStatus, TypeName } from '@appServices/apollo/status';
 
-type DatesSubscriptionCallback = EntitySubscriptionCallback<Datetime, AdditionalDateMenuOptions>;
+type DatesSubscriptionCallback = EntityActionsSubscriptionCb<Datetime, 'datetime'>;
 
 const useDatesActionMenuHandler = (): DatesSubscriptionCallback => {
-	return useCallback<DatesSubscriptionCallback>(
-		({ entityType, entity: date }, { registerMenuItem }, { dateMenuItemProps: menuItemProps }) => {
-			// although this is not needed
-			if (entityType !== 'datetime') {
-				return;
-			}
+	return useCallback<DatesSubscriptionCallback>(({ entityType, entity: date }, { registerMenuItem }) => {
+		// although this is not needed
+		if (entityType !== 'datetime') {
+			return;
+		}
 
-			registerMenuItem('editDate', () => <EditDateButton {...menuItemProps} />);
+		registerMenuItem('editDate', () => <EditDateButton />);
 
-			registerMenuItem('assignTickets', () => <AssignTicketsButton id={date.id} {...menuItemProps} />);
+		registerMenuItem('assignTickets', () => <AssignTicketsButton id={date.id} />);
 
-			registerMenuItem('deleteTicket', () => {
-				const { isLoaded } = useStatus();
-				/* Delete button should be hidden to avoid relational inconsistencies */
-				return isLoaded(TypeName.tickets) && <DeleteDateButton id={date.id} {...menuItemProps} />;
-			});
-		},
-		[]
-	);
+		registerMenuItem('deleteTicket', () => {
+			const { isLoaded } = useStatus();
+			/* Delete button should be hidden to avoid relational inconsistencies */
+			return isLoaded(TypeName.tickets) && <DeleteDateButton id={date.id} />;
+		});
```

Comment:
cool 

...


... but would it be better to wrap the registration in the `isLoaded` check?

ie:

```
const { isLoaded } = useStatus();
/* Delete button should be hidden to avoid relational inconsistencies */
if ( isLoaded(TypeName.tickets) ) {
	registerMenuItem('deleteTicket', () => <DeleteDateButton id={date.id} /> );
}
```

Reply:
Wrapping the whole registration will not work as it will run before the rendered menu item gets stored in the registry. The status check has to be delayed as much as possible.